### DISCUSSION TITLE
Zustand for React JS

### DISCUSSION THREAD

**Discussion by qwinza** [deaafrizal/forum#8] [Troubleshooting Room]
Minta sara temen", saya mau pake zustand di react cuman masih bingung, bingungnya tuh saya pake zustand ini tuh dalam kondisi apaa ya?

#3 #react

**Comment by deaafrizal** [comment]
simpelnya ketika lu sendiri udah susah maenin state. makanya tools macem zustand, redux dan lain2 disebut state management.

kapan maen state jadi susah? ketika apps udah lebih kompleks, banyak interaksi / side effect antara data dengan UI.

kembali lagi, tools macem ginian itu gak diwajibin untuk dipake. kalau butuh pake, kalau gak butuh gak usah dipaksain untuk dipake (kecuali dalam konteks belajar).

analogi simpelnya, ibarat beli iphone 15 tapi dipake cuma whatsapaan.

jadi semua kembali ke projeknya dan juga kebutuhannya. oh ya, dan kemampuan skill si programmernya juga. kalau sekedar pake tapi gak betul2 paham malah yang judul nya "manage state" bisa jadi "abstrak state" 🤣

semoga cukup terjawab
